File: Remove the balls.cpp
class Solution {
  public:
    int finLength(int N, vector<int> color, vector<int> radius) {
        
        stack<pair<int,int>> st;
        
        for (int i=0; i<N; i++) {
            pair<int, int> pii = {color[i], radius[i]};
            if (!st.empty() and st.top() == pii) st.pop();
            else st.push(pii);
        }
        
        return st.size();
    }
};

File: Remove the balls.java
class Solution {
    public static int finLength(int N, int[] color, int[] radius) {
        // code here
        Stack<Integer> st = new Stack<>();
        for(int i=0;i<N;i++){
            if(!st.isEmpty() && color[st.peek()]==color[i] && radius[st.peek()]==radius[i]){
                    st.pop();
                }
            else{
                st.push(i);
            }
        }
        return st.size();
    }
}
